### ISSUE TITLE
Dongle needs a power cycle if it loses connection to MQTT broker for extended period

### ISSUE THREAD

**Issue by danielraffel** [zakery292/monitormysolar#32]
If the dongle has an issue with the MQTT Broker (eg can't connect for an extended period of time) it seems to become unresponsive. Only fix is a physical power cycle.

https://github.com/user-attachments/assets/d87ae8ec-9c64-47cb-8171-11ff17e80369

**Comment by danielraffel** [comment]
<img width="173" alt="image" src="https://github.com/user-attachments/assets/e137a2fc-e6f3-487b-9b7b-c7a3af7b6dac">
interesting...just noticed the dongle portal says HA is "reconnecting" to HA even though it is definitely connected. I am seeing realtime data and it's updating.

![IMG_1AC733FCD158-1](https://github.com/user-attachments/assets/2bc5566a-8fb2-4cab-a184-0a5a2e511c8d)

**Comment by danielraffel** [comment]
the dongle is connecting to a mac mini running mosquitto which is relaying to my VM. here's the `mosquitto.conf`:

```
listener 1883 0.0.0.0
allow_anonymous true  # Use false if you want to require credentials for the dongle

# Bridge Configuration (Relay to VM over Tailscale)
connection tailscale_bridge
address tailscale_IP:1883
remote_username <username>
remote_password <22-digit password>
```

**Comment by danielraffel** [comment]
<img width="165" alt="image" src="https://github.com/user-attachments/assets/e8708a36-29f3-4f19-ac32-e41d05caf697">
Switched back to sending the MQTT data from the dongle directly to MQTT eclipse running on a VM in GCP. Something about connecting to a local machine and relaying via Mosquitto (over tailscale) isn't happy.

**Comment by danielraffel** [comment]
dongle has required reboot several times since returning to sending data without tailscale. I cannot connect to the portal. about to go unplug and plug it back in again. I dunno if this is due to the most recent firmware update but I did not see this happen before it.

**Comment by danielraffel** [comment]
Seems resolved with WiFi fixes in v2.0.32 firmware 👏
